### PR TITLE
Add docstrings for StoryGraph and StoryStep

### PR DESCRIPTION
**Pull request by msamogh**
**Proposed changes**:
- Add a brief description for the `StoryGraph` class

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

File: rasa/core/training/structures.py [hunk 1]
```diff
@@ -392,6 +392,9 @@ def dump_to_file(self, filename, flat=False, e2e=False):
 
 
 class StoryGraph(object):
+    """The directed acyclic graph (DAG) of story-steps pooled from all the
+        stories present in the training data."""
```
Comment (Ghostvv) [inline rasa/core/training/structures.py]: it's not acyclic, by default, there is a method that returns acyclic graph `with_cycles_removed`

File: rasa/core/training/structures.py [hunk 2]
```diff
@@ -92,6 +92,10 @@ def __repr__(self):
 
 
 class StoryStep(object):
+    """A StoryStep is a linear section of a story block that contains
+        no branching (either due to checkpoints or combining multiple
+        intents with "OR"s)."""
```
Comment (Ghostvv) [inline rasa/core/training/structures.py]: I think, it's a bit unclear, maybe rephrase it that it is the story block between checkpoints

Comment (msamogh) [reply rasa/core/training/structures.py]: I've updated it.

File: rasa/core/training/structures.py [hunk 3]
```diff
@@ -392,6 +399,9 @@ def dump_to_file(self, filename, flat=False, e2e=False):
 
 
 class StoryGraph(object):
+    """The graph of the story-steps pooled from all the stories
+        present in the training data."""
```
Comment (Ghostvv) [inline rasa/core/training/structures.py]:
```suggestion
    present in the training data."""
```

File: rasa/core/training/structures.py [hunk 4]
```diff
@@ -92,6 +92,13 @@ def __repr__(self):
 
 
 class StoryStep(object):
+    """A StoryStep is a section of a story block between two checkpoints.
+        
+        NOTE: Checkpoints are not only limited to those manually written
+        in the story file, but are also implicitly created at points where
+        multiple intents are separated in one line by chaining them with "OR"s.
```
Comment (Ghostvv) [inline rasa/core/training/structures.py]: there should be no tab

File: rasa/core/training/structures.py [hunk 5]
```diff
@@ -392,6 +399,10 @@ def dump_to_file(self, filename, flat=False, e2e=False):
 
 
 class StoryGraph(object):
+    """The graph of the story-steps pooled from all the stories
+    present in the training data.
```
Comment (Ghostvv) [inline rasa/core/training/structures.py]: there should we one sentence description that fits in one line then blank line then additional description then `"""` in the new line then blank line

Comment (msamogh) [reply rasa/core/training/structures.py]: Yeah, but I just have a one line description for now. So I just gotta move the triple quotes right after the period, right?

Comment (Ghostvv) [reply rasa/core/training/structures.py]: but it doesn't fit on one line

Comment (msamogh) [reply rasa/core/training/structures.py]: Does it have to? You're referring to something like this, right? https://github.com/RasaHQ/rasa/blob/f91a9caf6a04045fb6d86a4a04f4b2236f2df47b/rasa/core/featurizers.py#L23

Comment (Ghostvv) [reply rasa/core/training/structures.py]: oh, this is wrong

Comment (Ghostvv) [reply rasa/core/training/structures.py]: https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings

Comment (msamogh) [reply rasa/core/training/structures.py]: You're in for a refactoring treat, then 😄!

I came up with this regex to find such instances `:\n\h+""".*$\n\h+.*`, and I think there are around 286 matches.

Comment (msamogh) [reply rasa/core/training/structures.py]: Oh wait, a lot of false positives.

Comment (msamogh) [reply rasa/core/training/structures.py]: 45 matches (`:\n\h+""".*$\n\h+.*`)

Comment (msamogh) [reply rasa/core/training/structures.py]: What do you suggest we do?

Comment (Ghostvv) [reply rasa/core/training/structures.py]: I fixed featurizer.py in my core_transformer PR.

Comment (msamogh) [reply rasa/core/training/structures.py]: I've fixed mine. What about the other 43 instances, though?

Comment (Ghostvv) [reply rasa/core/training/structures.py]: you can create separate PR 😂

Comment (msamogh) [reply rasa/core/training/structures.py]: In my defence: "Some other good reasons to ignore a particular guideline:
- To be consistent with surrounding code that also breaks it"

(https://www.python.org/dev/peps/pep-0008/#a-foolish-consistency-is-the-hobgoblin-of-little-minds)

Comment (Ghostvv) [reply rasa/core/training/structures.py]: ok, but I rather change the docstrings
